force-app/main/default/classes/AccountTriggerHandlerTest.cls

```diff
@@ -0,0 +1,73 @@
+@isTest
+private class AccountTriggerHandlerTest {
+    @isTest static void testProcessBeforeInsertWithBypass() {
+        TriggerBypassUtility.enableBypassAccountTrigger();
+
+        Test.startTest();
+        Account testAccount = new Account(Name='Test Account', SLA__c=GlobalConsts.ACCOUNT_SLA_GOLD, CustomerPriority__c=GlobalConsts.ACCOUNT_PRIORITY_LOW);
+        insert testAccount;
+        Test.stopTest();
+
+        Account insertedAccount = [SELECT Id, Name FROM Account WHERE Id = :testAccount.Id];
+        System.assertNotEquals(null, insertedAccount);
+
+        TriggerBypassUtility.disableBypassAccountTrigger();
+    }
+
+    @isTest static void testProcessBeforeInsertWithBypassForUser() {
+        Id userId = UserInfo.getUserId();
+        TriggerBypassUtility.enableBypassTriggersForUser(userId);
+
+        Test.startTest();
+        Account testAccount = new Account(Name='Test Account', SLA__c=GlobalConsts.ACCOUNT_SLA_GOLD, CustomerPriority__c=GlobalConsts.ACCOUNT_PRIORITY_LOW);
+        insert testAccount;
+        Test.stopTest();
+
+        Account insertedAccount = [SELECT Id, Name FROM Account WHERE Id = :testAccount.Id];
+        System.assertNotEquals(null, insertedAccount);
+
+        TriggerBypassUtility.disableBypassTriggersForUser(userId);
+    }    
+
+    @isTest static void testProcessBeforeInsertWithoutBypass() {
+        Account testAccount = new Account(Name='Test Account', SLA__c=GlobalConsts.ACCOUNT_SLA_GOLD, CustomerPriority__c=GlobalConsts.ACCOUNT_PRIORITY_LOW);
+        
+        Test.startTest();
+        try {
+            insert testAccount;
+        } catch (DmlException e) {
+            System.assert(e.getMessage().contains(GlobalConsts.ACCOUNT_ERROR_MESSAGE_PRIORITY_BEFORE_INSERT));
```

Comment:
[MAJOR/DOUBT]
Jeśli nie wleci exception, nigdy nie sprawdzimy tej asercji. Tak powinno być?

Reply:
Przypisać do zmiennej message i assertować na końcu czy ten message się pojawił

Reply:
Try catch to najprostszy sposób na przechwycenie błędu. Istnieje możliwość spełnienia warunku po zmianach w kodzie, przez, które nastąpi brak przejścia do bloku catch - dlatego wrzucam zmienną. Można by jeszcze skorzystać z Database saveResult i przy niepowodzeniu sprawdzać error mesages i przeszukać tego, który ma wystąpić.